generator/templates/pack_unpack.hpp.j2

```diff
@@ -38,4 +35,31 @@
 
 #define BIT_GET(input, bit_idx) (bool)((input) & (1UL << (bit_idx)))
 
-#endif  // _CAN_LIBRARY_EVIL_MACROS_H
+struct DATA_T {
+  uint8_t byte[8];
+  uint64_t bitstring;
+};
+
+// utility functions
+void data_transfer(DATA_T *in, DATA_T *out);
+
+void to_bitstring(uint8_t in[], uint64_t *out);
+
+void from_bitstring(uint64_t *in, uint8_t out[]);
+
+uint8_t swap_uint8(uint8_t val);
+
+int8_t swap_int8(int8_t val);
+
+uint16_t swap_uint16(uint16_t val);
+
+int16_t swap_int16(int16_t val);
+
+uint32_t swap_uint32(uint32_t val);
+
+int32_t swap_int32(int32_t val);
+
+int64_t swap_int64(int64_t val);
+
+uint64_t swap_uint64(uint64_t val);
+
```

Comment:
This doesn't need to be a template, does it

Reply:
No, is this a problem?